benchmark/url/legacy-vs-whatwg-url-get-prop.js

```diff
@@ -0,0 +1,91 @@
+'use strict';
+const common = require('../common.js');
+const url = require('url');
+const URL = url.URL;
+const assert = require('assert');
+
+const inputs = {
+  long: 'http://nodejs.org:89/docs/latest/api/url.html#test?' +
+        'payload1=true&payload2=false&test=1&benchmark=3&' +
+        'foo=38.38.011.293&bar=1234834910480&test=19299&3992&' +
+        'key=f5c65e1e98fe07e648249ad41e1cfdb0',
+  short: 'https://nodejs.org/en/blog/',
+  idn: 'http://你好你好',
+  auth: 'https://user:pass@example.com/path?search=1',
+  special: 'file:///foo/bar/test/node.js'
+};
+
+const bench = common.createBenchmark(main, {
+  type: Object.keys(inputs),
+  method: ['legacy', 'whatwg'],
+  n: [1e5]
+});
+
+// At the time of writing, when using a passed property name to index
+// the object, Crankshaft would generate a LoadKeyedGeneric even when it
+// remains a constant in the function, so here we must use the literal
+// instead to get a LoadNamedField.
+function useLegacy(n, input) {
+  var obj = url.parse(input);
+  var noDead = {};
+  bench.start();
+  for (var i = 0; i < n; i += 1) {
+    noDead = {
```

Comment:
Yeah that would be better. I was not sure about this because some properties might be null, but come to think of it maybe I can just intialize them all to undefined.